packages/ember-auto-import/ts/bundle-config.ts

```diff
@@ -9,61 +9,70 @@ const testsPattern = new RegExp(`^/?[^/]+/(tests|test-support)/`);
 
 import type { TreeType } from './analyzer';
 
-function exhausted(value: never): never {
-  throw new Error(`Unknown tree type specified: ${value}`);
+function exhausted(label: string, value: never): never {
+  throw new Error(`Unknown ${label} specified: ${value}`);
 }
 
+export type BundleName = 'app' | 'tests';
+export type BundleType = 'js' | 'css';
+
 export default class BundleConfig {
   constructor(private emberApp: AppInstance) {}
 
   // This list of valid bundles, in priority order. The first one in the list that
   // needs a given import will end up with that import.
-  get names() : ReadonlyArray<string> {
+  get names() : ReadonlyArray<BundleName> {
     return Object.freeze(['app', 'tests']);
   }
 
-  get types(): ReadonlyArray<string> {
+  get types(): ReadonlyArray<BundleType> {
     return Object.freeze(['js', 'css']);
   }
 
   // Which final JS file the given bundle's dependencies should go into.
-  bundleEntrypoint(name: string, type: string): string | undefined {
+  bundleEntrypoint(name: BundleName, type: BundleType): string | undefined {
     switch (name) {
       case 'tests':
         switch (type) {
           case 'js':
             return 'assets/test-support.js';
           case 'css':
             return 'assets/test-support.css';
+          default:
+            exhausted('test bundle type', type);
         }
       case 'app':
         switch (type) {
           case 'js':
             return this.emberApp.options.outputPaths.vendor.js.replace(/^\//, '');
           case 'css':
             return this.emberApp.options.outputPaths.vendor.css.replace(/^\//, '');
+          default:
+            exhausted('app bundle type', type);
         }
+      default:
+        exhausted('bundle name', name);
     }
   }
 
-  bundleForTreeType(treeType: TreeType) {
+  bundleForTreeType(treeType: TreeType): BundleName {
     switch (treeType) {
       case 'app':
       case 'addon':
         return 'app';
 
       case 'addon-test-support':
       case 'test':
-        return 'test';
+        return 'tests';
```

Comment:
This is the real issue, it should have been "tests". The rest of the changes are around making the types fail compilation for this mistake.